components/ai_chat/browser/models.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include <vector>
+
+#include "brave/components/ai_chat/browser/models.h"
+
+#include "brave/components/ai_chat/common/mojom/ai_chat.mojom.h"
+
+namespace ai_chat {
+
+constexpr char kModelsDefaultKey[] = "chat-default";
+
+const base::flat_map<base::StringPiece, mojom::Model> kAllModels = {
```

Comment:
Can this be constructed at compile-time? i.e `base::MakeFlatMap`

Reply:
not for a mojom struct AFAIK unless you know of a way

Reply:
ah, I forget that Mojo types are not constexpr constructible. Their ctors need to be marked as constexpr but then ::New() methods will probably never be constexpr.